.github/workflows/test.yml

```diff
@@ -0,0 +1,158 @@
+
+on: pull_request
+
+jobs:
+
+  frontend:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        gulp_cmd:
+          - lint
+          - test:unit --ci --headless
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Node
+        uses: actions/setup-node@v1
+        with:
+            node-version: '10.x'
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+
+      - name: Install dependencies
+        run: |
+          ./frontend.sh development
+          python -m pip install --upgrade pip
+          pip install -r requirements/ci.txt
+
+      - name: Run ${{ matrix.gulp_cmd }}
+        run: yarn run gulp ${{ matrix.gulp_cmd }}
+
+      - name: Store test coverage
+        # Submit coverage from the unit test run only
+        if: matrix.gulp_cmd == 'test:unit --ci --headless'
+        uses: actions/upload-artifact@v1
+        with:
+          name: frontend_coverage
+          path: ./test/unit_test_coverage/clover.xml
+
+  backend:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        toxenv:
+          - lint-current
+          - unittest-current
+          # - unittest-future
+
+    services:
+      postgres:
+        # https://github.com/actions/example-services/blob/master/.github/workflows/postgres-service.yml
+        image: postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        ports:
+        - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements/ci.txt
+
+      - name: Run tox -e ${{ matrix.toxenv }}
+        run: |
+          tox -e ${{ matrix.toxenv }}
+        env: 
+          TEST_RUNNER: cfgov.test.StdoutCapturingTestRunner 
+          TEST_DATABASE_URL: postgres://postgres:postgres@localhost/postgres
+          CI_CHECK_MIGRATIONS: "./manage.py makemigrations --dry-run --check"
+
+      - name: Prepare test coverage
+        if: matrix.toxenv == 'unittest-current'
+        run: coverage xml
+
+      - name: Store test coverage
+        # Submit coverage from our prefered tox run only
+        if: matrix.toxenv == 'unittest-current'
+        uses: actions/upload-artifact@v1
+        with:
+          name: backend_coverage
+          path: ./coverage.xml
+
+  frontend-coverage:
+    runs-on: ubuntu-latest
+    needs: 
+      - frontend
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
```

Comment:
Why do you need to do this instead of the default `actions/checkout` behavior?

Reply:
Because the `diff-cover` tool compares against `master`, so `master` needs to exist. It does not with the default `actions/checkout` behavior.

Reply:
Cool.  Makes sense.